File: CleanUpTheWords/c/cutw.c
// clean up the words soln in c for code eval by steven a dunn

#include <stdio.h>

int main (int argc, char* const argv[])
{
	FILE *fp = fopen(argv[1], "r");
	if (fp != NULL)
	{
		char line[1024];
		while (fgets(line, sizeof(line), fp))
		{
			int i;
			int print_space = 0;
			
			for (i = 0; i < sizeof(line); i++) {
				if (line[i] == '\n' || line[i] == '\0')
					break;

				if (isalpha(line[i]))
				{
					printf("%c", tolower(line[i]));
					print_space = 1;
				}
				else {
					if (print_space)
					{
						printf(" ");
						print_space = 0;
					}
				}
			}
			puts("");
		}
		fclose(fp);
	}
	
	
	return 0;
}

File: CleanUpTheWords/py3/cutw.py
# clean up the words soln in py3 for code eval by steven a dunn

import sys, re

f = open(sys.argv[1], 'r')
for line in f:
	words = re.compile('[A-Za-z]+').findall(line)
	for word in words:
		print (word.lower(), end=' ')
	print()
f.close()